source/Subsystem/EntityComponentSystem.h

```diff
@@ -55,9 +59,32 @@ class EntityComponentSystem
 		return ComponentType::UNDEFINED;
 	}
 
+	UniqueID AddEntity()
+	{
+		Entity* entity = new Entity;
+		myEntityTable.emplace(myUIDCounter, entity);
+		myUIDCounter += 1;
+		return myUIDCounter - 1;
+	}
+
+	Entity* GetEntity(UniqueID aUID)
+	{
+		return myEntityTable[aUID];
+	}
+
+	void RemoveEntity(UniqueID aUID)
+	{
+		if (!myEntityTable[aUID])
+			return;
+		
+		delete myEntityTable[aUID];
+		myEntityTable[aUID] = nullptr;
```

Comment:
Why not store the result out of myEntityTable[UID] once and use it for the rest of the function.